exercises/pangram/instructions.md

```diff
@@ -5,4 +5,4 @@ Your task is to figure out if a sentence is a pangram.
 A pangram is a sentence using every letter of the alphabet at least once.
 It is case insensitive, so it doesn't matter if a letter is lower-case (e.g. `k`) or upper-case (e.g. `K`).
 
-For this exercise we only use the basic letters used in the English alphabet: `a` to `z`.
+For this exercise we only consider the basic letters used in English as part of the alphabet: `a` to `z`.
```

Comment:
Maybe we should be more explicit, eg Note: non-ASCII characters should not be considered as containing the letters needed to meet the pangram requirements.

Reply:
I don't see how that's less ambiguous than my first suggestion ("a sentence is a pangram if it contains each of the 26 basic characters in the English alphabet") because neither specify what you should do about characters that are followed by combining diacritcs.

I think it's probably fine to not be more specific, though. Probably most students won't think about diacritics and most tracks won't include any tests including them. (The Julia track currently does, but maybe we will change that).

I kinda want something that is clear without requiring the student to know about or think about charsets.

Reply:
The issue is that identifying whether a character has diacritics or not is very difficult, as I explained earlier. If you say "non-ASCII characters should not be considered as containing the letters needed to meet the pangram requirements." then does that mean that you can ignore all non-ascii bytes or that you need to check if each ascii letter is followed by any combining diacritic marks?

As argued earlier, you can't solve the problem in the presence of diacritics without unicode normalisation (unless you are happy with visually-identical texts having different results).

So to be fully precise you have to say that the whole input is ASCII, or that there are no combining diacritic characters (or only combining diacritics and no composed diacritics or vice versa). Saying that the input is ASCII only is a bit rubbish (imho) because it reinforces the assumption that strings are ASCII, but it also seems like a bad idea to encourage the student to get into the weeds of unicode string identity.

I think most students won't think about diacritics, so we can just not mention them. Tracks that want to include unicode tests can use non-latin characters to sidestep the issue. If we think students will think about it, then this is my suggestion for a slightly more detailed description:

> For this exercise, a sentence is a pangram if it contains each of the 26 letters in the English alphabet. Your input will not contain accents or diacritics.

I also mentioned the issues around whether café contains 3 or 4 english letters earlier. Did you read my full comment?

Reply:
> Maybe a French or German speaker would say that e and é and è are three different letters (and how about groß, does that contain s? And does cœur contain o?).

Groß does not contain s, except in Switzerland. äöü are their own letters.

> I think this is clearer:

Agreed.

Reply:
> I kinda want something that is clear without requiring the student to know about or think about charsets.

Yeah that was my intention as well. I want to make it clear that the input isn't restricted to 'a':'z' even if only those matter for determining whether the input is a pangram.